From 66d285f6525c355492d00389e8efb3234be36a75 Mon Sep 17 00:00:00 2001
From: Kevin Strasser <kevin.strasser@linux.intel.com>
Date: Wed, 24 Oct 2012 18:15:39 -0700
Subject: [PATCH 1/3] internals.py: wait forever for searial reads

this will stop the listener thread from busy waiting when there is
no data on the serial port.
---
 robovero/internals.py |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/robovero/internals.py b/robovero/internals.py
index bde65a1..0222a93 100644
--- a/robovero/internals.py
+++ b/robovero/internals.py
@@ -20,10 +20,8 @@ def listen():
           
     if message == "":
       robovero.debug.write("[%f] INTERRUPT: " % (time.time() - robovero.start_time))
-      robovero.serial.timeout = None
       IRQn = int(robovero.readline("\r\n"), 16)
       robovero.debug.write("%x\r\n" % IRQn)
-      robovero.serial.timeout = 0
       isrthread = threading.Thread(target=isr, args=[IRQn], name="isr")
       isrthread.start()
       
@@ -213,7 +211,7 @@ class Robovero(object):
       exit("Couldn't open device.")
 
     self.start_time = time.time()
-    self.serial.timeout = 0
+    self.serial.timeout = None
     # send line terminator, disable console echo and prompt
     self.serial.write("\r\n")
     self.serial.write("promptOff\r\n")
-- 
1.7.9.5

